src/UNL/Knowledge/Records.php

```diff
@@ -124,4 +135,108 @@ protected function getMergedKey($item, $section, $tags)
 
     	return implode(' ', $results);
     }
+
+    protected function getPublicProperties()
+    {
+        $self = $this;
+        $getPublicProperties = function() use ($self) {
+            return get_object_vars($self);
+        };
+        $getPublicProperties = $getPublicProperties->bindTo(null, null);
+
+        return $getPublicProperties();
```

Comment:
Of course it has use cases, otherwise it wouldn't exist. In general, _reverse engineering_ something seems like _over_ engineering something that probably should have been more explicit in an API. 

I also see it used to read phpDoc or other class comments to implement class annotations. I am not a fan of storing program data in comments and know that opcache has an option to remove comments from the cache storage.